File: java/sprint2/D/Solution.java
// <template>
class Node<V> {  
    public V value;  
    public Node<V> next;  
 
    public Node(V value, Node<V> next) {  
        this.value = value;  
        this.next = next;  
    }  
}
// <template>

public class Solution {
    public static int solution(Node<String> head, String elem) {
        // Your code
        // ヽ(´▽`)/
    }

    private static void test() {
        Node<String> node3 = new Node<>("node3", null);
        Node<String> node2 = new Node<>("node2", node3);
        Node<String> node1 = new Node<>("node1", node2);
        Node<String> node0 = new Node<>("node0", node1);
        int idx = solution(node0, "node2");
        assert idx == 2;
    }
}

File: kotlin/sprint2/D/Solution.kt
// <template>
class Node<V>(var value: V, var next: Node<V>?) {}
// <template>

fun solution(head: Node<String>?, item: String): Int {
    // Your code
    // ヽ(´▽`)/
}

fun test() {
    val node3 = Node("node3", null)
    val node2 = Node("node2", node3)
    val node1 = Node("node1", node2)
    val node0 = Node("node0", node1)
    val idx: Int = solution(node0, "node2")
    assert(idx == 2)
}